gossip/server.go

```diff
@@ -88,6 +88,10 @@ func (s *server) Gossip(argsI proto.Message) (proto.Message, error) {
 	if err != nil {
 		return nil, util.Errorf("local addr %s could not be converted to net.Addr: %s", args.LAddr, err)
 	}
+	// Verify that the client connection is valid and hasn't been closed.
+	if _, ok := s.lAddrMap[lAddr.String()]; !ok {
```

Comment:
I think `args.LAddr` is set by the sender to its local port, so it should be the other guy's ephemeral port, and thus match your description of `conn.RemoteAddr`.